src/win.rs

```diff
@@ -171,6 +168,38 @@ impl<T: Write + Send> WinConsole<T> {
     }
 }
 
+impl<T: Write + Send> WinConsole<T> {
+    fn apply(&mut self) -> io::Result<()> {
+        let out = conout()?;
+        let _unused = self.buf.flush();
+        let mut accum: WORD = 0;
+        accum |= color_to_bits(self.foreground);
+        accum |= color_to_bits(self.background) << 4;
+        unsafe {
+            SetConsoleTextAttribute(*out, accum);
+        }
+        Ok(())
+    }
+
+    /// Create a new WinConsole with the given WinConsoleInfo and out
+    pub fn new_with_consoleinfo(out: T, consoleinfo: WinConsoleInfo) -> WinConsole<T> {
+        WinConsole {
+            buf: out,
+            def_foreground: consoleinfo.def_foreground,
```

Comment:
I'd modify the `WinConsole` struct itself to embed a `WinConsoleInfo`.

Reply:
Right. I thought a smaller diff would be better.

Are you okay the idea behind the change?

Reply:
Definitely. This brings the windows/unix interfaces closer together without breaking the API in any way.

Reply:
Done.

If you ever have the time. You feedback about [term-string](https://github.com/rust-alt/term-string) would be greatly appreciated.

Reply:
I'll take a look when I get a chance.